src/main/java/org/opensrp/web/rest/LocationResource.java

```diff
@@ -346,6 +324,10 @@ public ResponseEntity<String> getStructuresWithinCordinates(@RequestParam(value
 	 * @param isJurisdiction    boolean which when true the search is done on jurisdictions and when false search is on structures
 	 * @param returnGeometry    boolean which controls if geometry is returned
 	 * @param propertiesFilters list of params with each param having name and value e.g name:House1
+	 * Amend to accept parentId in the following formats;
+	 * 1. parentId:
+	 * 2. parentId:""
+	 * 3. parentId:null
```

Comment:
@SebaMutuku what happens if the location doesn't have a `parentId`. This location should also be treated as root

Reply:
@dubdabasoduba , I don't believe this is  a necessary check since:
1. All locations in the database either have an empty parent or a value. But the column for parentId exists hence writing queries to check for missing parentId column will always be false.
2. The opensrp documentation says the location to be updated as root should be in the format `  {
    "type": "Feature",
    "id": "02ebbc84-5e29-4cd5-9b79-c594058923e9",
    "properties": {
      "status": "Active",
      "parentId": "",
      "name": "Uganda",
      "geographicLevel": 0,
      "version": 0
    },
    "serverVersion": 1,
    "locationTags": [
      {
        "id": 1,
        "name": "Country"
      }
    ]
  }`
.This means the parentId column key will always be present.